core/tx_pool.go

```diff
@@ -1118,6 +1121,15 @@ func (pool *TxPool) reset(oldHead, newHead *types.Header) {
 	log.Debug("Reinjecting stale transactions", "count", len(reinject))
 	senderCacher.recover(pool.signer, reinject)
 	pool.addTxsLocked(reinject, false)
+
+	// Update all fork indicators by next pending block number.
+	next := new(big.Int).Add(newHead.Number, big.NewInt(1))
+	pool.istanbul = pool.chainconfig.IsIstanbul(next)
+	if pool.istanbul {
+		pool.homestead = true
+	} else {
+		pool.homestead = pool.chainconfig.IsHomestead(next)
+	}
```

Comment:
If the dynasty is changed, should we recheck all pending txs which we receive by old consensus rules?

Reply:
No, no need to recheck past ones. We only care that the intrinsic gas is lower than the total requested gas when inserting into the pool. As we're pushing the gas down, it will always hold true when transitioning into Istanbul.

If we have a reorg back from Istanbul into Homestead that might be a bit messy indeed, since we might accept some transactions that would become invalid, but I don't think this is a legit issue we need to concern ourselves with. **At worse** the pool might be shuffling junk until the chain progresses again into Istanbul.